### ISSUE TITLE
Recover from wrong git credentials during update

### ISSUE THREAD

**Issue by barredterra** [frappe/bench#1268]
Issue: Feature Request

**Is your feature request related to a problem? Please describe.**
When running `bench update` with some private custom apps, the `git fetch` part asks for username and password. When there is a typo in the credentials, the entire `bench update` fails immediately.

I can't count the times I had to re-run `bench update` because I messed up the credentials.

**Describe the solution you'd like**

- Retry the failed `git fetch` until the user explicitly cancels the process
- Optional: a way to pass git credentials for an unattended update

**Comment by ankush** [comment]
Not a solution but use SSH keys man, using user/pass is very tiring 😄

**Comment by barredterra** [comment]
@ankush you're right. In some cases my problem could be avoided by using [Deploy Keys](https://docs.github.com/en/developers/overview/managing-deploy-keys#deploy-keys). But it looks like that requires a complicated setup to work reliably at scale: many organizations, many repos, varying permissions (can read but cannot set deploy keys), many servers, ....
